cdap-app-fabric/src/test/java/io/cdap/cdap/internal/sysapp/SystemAppManagementServiceTest.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.sysapp;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.inject.Injector;
+import io.cdap.cdap.AllProgramsApp;
+import io.cdap.cdap.api.artifact.ArtifactSummary;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.id.Id;
+import io.cdap.cdap.internal.app.services.ApplicationLifecycleService;
+import io.cdap.cdap.internal.app.services.ProgramLifecycleService;
+import io.cdap.cdap.internal.app.services.http.AppFabricTestBase;
+import io.cdap.cdap.proto.ProgramType;
+import io.cdap.cdap.proto.artifact.AppRequest;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.id.ProgramId;
+import org.iq80.leveldb.util.FileUtils;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests for the {@link SystemAppManagementService}.
+ */
+public class SystemAppManagementServiceTest extends AppFabricTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SystemAppManagementServiceTest.class);
+  private static final Gson GSON = new Gson();
+
+  private static ProgramLifecycleService programLifecycleService;
+  private static ApplicationLifecycleService applicationLifecycleService;
+  private static CConfiguration cConf;
+  private static SystemAppManagementService systemAppManagementService;
+  private static File systemConfigDir;
+
+  private static final String RUNNING = "RUNNING";
+
+  @ClassRule
+  public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    Injector injector = getInjector();
+    programLifecycleService = injector.getInstance(ProgramLifecycleService.class);
+    applicationLifecycleService = injector.getInstance(ApplicationLifecycleService.class);
+    cConf = injector.getInstance(CConfiguration.class);
+    systemConfigDir = TEMPORARY_FOLDER.newFolder("demo-sys-app-config-dir");
+    cConf.set(Constants.SYSTEM_APP_CONFIG_DIR, systemConfigDir.getAbsolutePath());
+    systemAppManagementService = new SystemAppManagementService(cConf, applicationLifecycleService,
```

Comment:
Since the service can't really be re-used across test cases because it reads things on start up, it's better to create it within the test case rather than here.

Reply:
Makes sense. I was planning to use @BeforeTest and @AfterTest but this looks better for now. Changed it accordingly.